### PR TITLE
Save/Load the layout settings of Panels is supported.

### PR DESCRIPTION
**Pull request by Vane11ope**
gifrecord_2018-09-04_035703

### CODE REVIEW

**Comment by Vane11ope** [comment]
will fix this tomorrow

File: libr/core/panels.c [hunk 1]
```diff
@@ -2148,6 +2163,73 @@ static void restorePanelPos(RPanel* panel) {
...
+	for (i = 0; i < panels->n_panels; i++) {
+		RPanel *panel = &panels->panel[i];
+		RJSVar* obj = r_json_object_new ();
+		(void)r_json_object_add (obj, "Title", r_json_string_new (panel->title));
+		(void)r_json_object_add (obj, "Cmd", r_json_string_new (panel->cmd));
+		(void)r_json_object_add (obj, "x", r_json_number_new (panel->pos.x));
+		(void)r_json_object_add (obj, "y", r_json_number_new (panel->pos.y));
+		(void)r_json_object_add (obj, "w", r_json_number_new (panel->pos.w));
+		(void)r_json_object_add (obj, "h", r_json_number_new (panel->pos.h));
+		char* c = r_json_stringify (obj, true);
```
Comment (radare) [inline libr/core/panels.c]: leaked string and leaked json object

Comment (radare) [reply libr/core/panels.c]: why not use k=v in sdb?

File: libr/core/panels.c [hunk 2]
```diff
@@ -2148,6 +2163,73 @@ static void restorePanelPos(RPanel* panel) {
 	panel->pos.h = panel->prevPos.h;
 }
 
+static void savePanelsLayout(RPanels* panels) {
+	const char *configPath = r_str_home (".r2panels");
```
Comment (radare) [inline libr/core/panels.c]: will be good to honor to use the XDG dirctories instead of ~/.r2panels

File: libr/core/panels.c [hunk 3]
```diff
@@ -2148,6 +2163,73 @@ static void restorePanelPos(RPanel* panel) {
...
+	RPanels *panels = core->panels;
+	const char *configPath = r_str_home (".r2panels");
+	int i;
+	if (!configPath){
+		return;
+	}
+	panelAllClear (panels);
+	panels->n_panels = 0;
+	panels->curnode = 0;
+	FILE *panelsConfig = fopen (configPath, "r");
```
Comment (radare) [inline libr/core/panels.c]: check if fopen succeeds or not

File: libr/core/panels.c [hunk 4]
```diff
@@ -2148,6 +2163,73 @@ static void restorePanelPos(RPanel* panel) {
...
+	int max = 256;
+	char l[max];
+	char *title, *cmd, *x, *y, *w, *h;
+	while (fgets (l, max, panelsConfig)) {
```
Comment (radare) [inline libr/core/panels.c]: last ine will be executed twice

Comment (radare) [inline libr/core/panels.c]: use r_file apis to avoid accessing files outside the sandbox

Comment (radare) [reply libr/core/panels.c]: you can just r_file_slurp() and the r_str_split("\n") or use strchr (\n)

**Comment by radare** [comment]
doesnt builds on windows

**Pull request merged**